rye/src/bootstrap.rs

```diff
@@ -547,6 +466,187 @@ pub fn download_url_ignore_404(url: &str, output: CommandOutput) -> Result<Optio
     }
 }
 
+// Represents a uv binary and associated functions
+// to bootstrap rye using uv.
+#[derive(Clone)]
+struct Uv {
+    output: CommandOutput,
+    uv_bin: PathBuf,
+}
+
+impl Uv {
+    // Ensure we have a uv binary for bootstrapping
+    fn ensure_exists(output: CommandOutput) -> Result<Self, Error> {
+        // Request a download for the default uv binary for this platform.
+        // For instance on aarch64 macos this will request a compatible uv version.
+        let download = UvDownload::try_from(UvRequest::default())?;
+        let uv_dir = get_app_dir().join("uv").join(download.version());
```

Comment:
Is there much of a benefit of having multiple uv versions and not just to retain only the latest?

Reply:
The main reason here is that i can check for the version with one simple `stat` call and not have to `execv` uv. This allows me in later changes to make use of `ensure_exists` to obtain a `Uv` on any code path. It also opens the door for pinning uv if we ever need to (I don't see why, but it feels more future proof). I am happy to just nuke the uv directory before I download a new version which will keep it clean for now. If you insist i can remove the version.